# **Analisis Pola dan Tren Perilaku Belanja Konsumen Berdasarkan Data Transaksi**

Nama: Yahya Jailani

Kelas: C

NIM: 2509116085

# **Import Library**

In [1]:
import pandas as pd

# **Memuat Dataset**

In [3]:
file = '/content/shopping_behavior_updated (1).csv'
df = pd.read_csv(file)

In [ ]:
df

Customer ID  Age  Gender Item Purchased     Category  \
0               1   55    Male         Blouse     Clothing   
1               2   19    Male        Sweater     Clothing   
2               3   50    Male          Jeans     Clothing   
3               4   21    Male        Sandals     Footwear   
4               5   45    Male         Blouse     Clothing   
...           ...  ...     ...            ...          ...   
3895         3896   40  Female         Hoodie     Clothing   
3896         3897   52  Female       Backpack  Accessories   
3897         3898   46  Female           Belt  Accessories   
3898         3899   44  Female          Shoes     Footwear   
3899         3900   52  Female        Handbag  Accessories   

      Purchase Amount (USD)       Location Size      Color  Season  \
0                        53       Kentucky    L       Gray  Winter   
1                        64          Maine    L     Maroon  Winter   
2                        73  Massachusetts    S     Maroon  Spring   
3                        90   Rhode Island    M     Maroon  Spring   
4                        49         Oregon    M  Turquoise  Spring   
...                     ...            ...  ...        ...     ...   
3895                     28       Virginia    L  Turquoise  Summer   
3896                     49           Iowa    L      White  Spring   
3897                     33     New Jersey    L      Green  Spring   
3898                     77      Minnesota    S      Brown  Summer   
3899                     81     California    M      Beige  Spring   

      Review Rating Subscription Status Discount Applied  Previous Purchases  \
0               3.1                 Yes              Yes                  14   
1               3.1                 Yes              Yes                   2   
2               3.1                 Yes              Yes                  23   
3               3.5                 Yes              Yes                  49   
4               2.7                 Yes              Yes                  31   
...             ...                 ...              ...                 ...   
3895            4.2                  No               No                  32   
3896            4.5                  No               No                  41   
3897            2.9                  No               No                  24   
3898            3.8                  No               No                  24   
3899            3.1                  No               No                  33   

     Payment Method Frequency of Purchases  
0             Venmo            Fortnightly  
1              Cash            Fortnightly  
2       Credit Card                 Weekly  
3            PayPal                 Weekly  
4            PayPal               Annually  
...             ...                    ...  
3895          Venmo                 Weekly  
3896  Bank Transfer              Bi-Weekly  
3897          Venmo              Quarterly  
3898          Venmo                 Weekly  
3899          Venmo              Quarterly  

[3900 rows x 16 columns]

In [6]:
df.head(10)

Customer ID  Age Gender Item Purchased     Category  Purchase Amount (USD)  \
0            1   55   Male         Blouse     Clothing                     53   
1            2   19   Male        Sweater     Clothing                     64   
2            3   50   Male          Jeans     Clothing                     73   
3            4   21   Male        Sandals     Footwear                     90   
4            5   45   Male         Blouse     Clothing                     49   
5            6   46   Male       Sneakers     Footwear                     20   
6            7   63   Male          Shirt     Clothing                     85   
7            8   27   Male         Shorts     Clothing                     34   
8            9   26   Male           Coat    Outerwear                     97   
9           10   57   Male        Handbag  Accessories                     31   

        Location Size      Color  Season  Review Rating Subscription Status  \
0       Kentucky    L       Gray  Winter            3.1                 Yes   
1          Maine    L     Maroon  Winter            3.1                 Yes   
2  Massachusetts    S     Maroon  Spring            3.1                 Yes   
3   Rhode Island    M     Maroon  Spring            3.5                 Yes   
4         Oregon    M  Turquoise  Spring            2.7                 Yes   
5        Wyoming    M      White  Summer            2.9                 Yes   
6        Montana    M       Gray    Fall            3.2                 Yes   
7      Louisiana    L   Charcoal  Winter            3.2                 Yes   
8  West Virginia    L     Silver  Summer            2.6                 Yes   
9       Missouri    M       Pink  Spring            4.8                 Yes   

  Discount Applied  Previous Purchases Payment Method Frequency of Purchases  
0              Yes                  14          Venmo            Fortnightly  
1              Yes                   2           Cash            Fortnightly  
2              Yes                  23    Credit Card                 Weekly  
3              Yes                  49         PayPal                 Weekly  
4              Yes                  31         PayPal               Annually  
5              Yes                  14          Venmo                 Weekly  
6              Yes                  49           Cash              Quarterly  
7              Yes                  19    Credit Card                 Weekly  
8              Yes                   8          Venmo               Annually  
9              Yes                   4           Cash              Quarterly

In [7]:
df.tail(10)

Customer ID  Age  Gender Item Purchased     Category  \
3890         3891   35  Female          Shirt     Clothing   
3891         3892   36  Female          Dress     Clothing   
3892         3893   35  Female        Jewelry  Accessories   
3893         3894   21  Female            Hat  Accessories   
3894         3895   66  Female          Skirt     Clothing   
3895         3896   40  Female         Hoodie     Clothing   
3896         3897   52  Female       Backpack  Accessories   
3897         3898   46  Female           Belt  Accessories   
3898         3899   44  Female          Shoes     Footwear   
3899         3900   52  Female        Handbag  Accessories   

      Purchase Amount (USD)       Location Size      Color  Season  \
3890                     81       Nebraska   XL      Green  Winter   
3891                     30       Colorado    L      Peach  Winter   
3892                     86       Michigan    L     Indigo  Summer   
3893                     64  Massachusetts    L      White    Fall   
3894                     78    Connecticut    L      White  Spring   
3895                     28       Virginia    L  Turquoise  Summer   
3896                     49           Iowa    L      White  Spring   
3897                     33     New Jersey    L      Green  Spring   
3898                     77      Minnesota    S      Brown  Summer   
3899                     81     California    M      Beige  Spring   

      Review Rating Subscription Status Discount Applied  Previous Purchases  \
3890            2.6                  No               No                  33   
3891            4.7                  No               No                   6   
3892            3.5                  No               No                   5   
3893            3.3                  No               No                  29   
3894            3.9                  No               No                  44   
3895            4.2                  No               No                  32   
3896            4.5                  No               No                  41   
3897            2.9                  No               No                  24   
3898            3.8                  No               No                  24   
3899            3.1                  No               No                  33   

     Payment Method Frequency of Purchases  
3890     Debit Card               Annually  
3891  Bank Transfer              Quarterly  
3892         PayPal            Fortnightly  
3893  Bank Transfer              Bi-Weekly  
3894    Credit Card         Every 3 Months  
3895          Venmo                 Weekly  
3896  Bank Transfer              Bi-Weekly  
3897          Venmo              Quarterly  
3898          Venmo                 Weekly  
3899          Venmo              Quarterly

# **Google Drive**

In [4]:
from google.colab import drive
drive.mount('/content/drive')

path = '/content/drive/MyDrive/Praktikum Season 2/shopping_behavior_updated (1).csv'

Mounted at /content/drive


In [5]:
df = pd.read_csv(path)
df

Customer ID  Age  Gender Item Purchased     Category  \
0               1   55    Male         Blouse     Clothing   
1               2   19    Male        Sweater     Clothing   
2               3   50    Male          Jeans     Clothing   
3               4   21    Male        Sandals     Footwear   
4               5   45    Male         Blouse     Clothing   
...           ...  ...     ...            ...          ...   
3895         3896   40  Female         Hoodie     Clothing   
3896         3897   52  Female       Backpack  Accessories   
3897         3898   46  Female           Belt  Accessories   
3898         3899   44  Female          Shoes     Footwear   
3899         3900   52  Female        Handbag  Accessories   

      Purchase Amount (USD)       Location Size      Color  Season  \
0                        53       Kentucky    L       Gray  Winter   
1                        64          Maine    L     Maroon  Winter   
2                        73  Massachusetts    S     Maroon  Spring   
3                        90   Rhode Island    M     Maroon  Spring   
4                        49         Oregon    M  Turquoise  Spring   
...                     ...            ...  ...        ...     ...   
3895                     28       Virginia    L  Turquoise  Summer   
3896                     49           Iowa    L      White  Spring   
3897                     33     New Jersey    L      Green  Spring   
3898                     77      Minnesota    S      Brown  Summer   
3899                     81     California    M      Beige  Spring   

      Review Rating Subscription Status Discount Applied  Previous Purchases  \
0               3.1                 Yes              Yes                  14   
1               3.1                 Yes              Yes                   2   
2               3.1                 Yes              Yes                  23   
3               3.5                 Yes              Yes                  49   
4               2.7                 Yes              Yes                  31   
...             ...                 ...              ...                 ...   
3895            4.2                  No               No                  32   
3896            4.5                  No               No                  41   
3897            2.9                  No               No                  24   
3898            3.8                  No               No                  24   
3899            3.1                  No               No                  33   

     Payment Method Frequency of Purchases  
0             Venmo            Fortnightly  
1              Cash            Fortnightly  
2       Credit Card                 Weekly  
3            PayPal                 Weekly  
4            PayPal               Annually  
...             ...                    ...  
3895          Venmo                 Weekly  
3896  Bank Transfer              Bi-Weekly  
3897          Venmo              Quarterly  
3898          Venmo                 Weekly  
3899          Venmo              Quarterly  

[3900 rows x 16 columns]

# **Business Understanding**

Perusahaan perlu memahami pola perilaku belanja konsumen dari data transaksi yang tersedia untuk mengoptimalkan strategi pemasaran dan meningkatkan nilai transaksi. Analisis ini bertujuan mengidentifikasi minimal tiga pola signifikan dan menghasilkan rekomendasi strategis yang dapat meningkatkan rata-rata nilai transaksi sebesar 8 sampai dengan 10% dalam enam bulan. Fokus analisis mencakup pengaruh kependudukan terhadap pola pembelian, tren musiman dan geografis, serta faktor-faktor utama yang mendorong nilai transaksi dan kepuasan pelanggan.



## **1. Business Objective**

1. Tujuan Bisnis: Memahami pola dan tren perilaku belanja konsumen untuk mengoptimalkan strategi pemasaran dan meningkatkan nilai transaksi.
2. Kriteria Kesuksesan:

*  Teridentifikasinya 3 pola/tren perilaku belanja signifikan dari data transaksi.

*  Dihasilkannya rekomendasi strategis yang dapat ditindaklanjuti.

*  Meningkatnya rata-rata nilai transaksi sebesar 8 hingga 10% dalam 6 bulan pasca implementasi.

Pertanyaan Bisnis yang digunakan:

1. Bagaimana pola pembelian bervariasi berdasarkan usia dan jenis kelamin?

2. Bagaimana tren pembelian berubah berdasarkan musim?

3. Apakah ada pola musiman untuk kategori produk tertentu?

4. Apa korelasi antara diskon, status langganan, dan metode pembayaran dengan nilai transaksi serta kepuasan pelanggan?

## **2. Asses Situation**

Inventarisasi Sumber Daya:

* Data: Dataset dengan 3.900 entri pelanggan, mencakup variabel: 'Customer ID, Age, Gender, Item Purchased, Category, Purchase Amount (USD), Location, Size, Color, Season, Review Rating, Subscription Status, Discount Applied, Previous Purchases, Payment Method, Frequency of Purchases'.

* Alat Analitik: Tools untuk analisis data seperti Python (Pandas, Scikit-learn), R, atau platform analisis visual seperti Tableau.

* Keahlian: Tim data scientist atau analis data yang mampu melakukan analisis eksplorasi data, segmentasi pelanggan, analisis korelasi dan pemodelan.

* Pemangku Kepentingan: Manajer produk, tim pemasaran, tim pengembangan algoritma rekomendasi, dan tim strategi bisnis.

Kerangka konseptual ini dapat digunakan untuk mengidentifikasi pola perilaku konsumen serta merumuskan strategi pemasaran yang tepat hingga pada akhirnya meningkatkan retensi dan nilai transaksi pelanggan secara berkelanjutan.

## **3. Analytic Goals & Project Plan**

Berdasarkan tujuan bisnis, tujuan analitik dirumuskan sebagai berikut:

1. Peluang Perusahaan: Membuka peluang untuk perusahaan dalam memasarkan produk dengan waktu yang tepat mengikuti perkembangan tren dan alur sosial media seperti Tiktok, Instagram, DLL.

2. Prediksi: Mengembangkan model untuk memprediksi kelompok usia atau preferensi perangkat berdasarkan pola penggunaan.

3. Rekomendasi Strategi: Memberikan rekomendasi spesifik untuk penargetan algoritma dan personalisasi konten untuk setiap kalangan.

Analisis ini juga relevan bagi pemerintah atau pengelola platform media sosial. Wawasan yang sama dapat digunakan untuk kampanye sosial positif, namun berpotensi disalahgunakan untuk membentuk opini publik atau mengeksploitasi kelompok tertentu.

Projek Plan untuk bisnis:

### Fase 1: Persiapan dan Pemahaman Data
* Mengumpulkan dan memvalidasi data dari file shopping_behavior_updated (1).csv.
* Melakukan eksplorasi data awal (EDA) untuk memahami distribusi, missing values, dan outliers.
* Mendefinisikan metrik evaluasi yang sejalan dengan kriteria kesuksesan bisnis.

### Fase 2: Analisis Eksploratori dan Segmentasi
* Analisis statistik deskriptif untuk setiap kelompok usia. Visualisasi pola penggunaan (misalnya, rata-rata jumlah pembelian per kategori produk per kelompok usia).
* Melakukan analisis klaster (clustering) untuk mengidentifikasi segmen pengguna yang homogen di dalam atau antar kelompok usia.

### Fase 3: Pemodelan dan Analisis Lanjutan
* Mengembangkan model klasifikasi untuk memprediksi kelompok usia berdasarkan variabel penggunaan.
* Uji statistik untuk memvalidasi perbedaan signifikan antar kelompok.